### PR TITLE
crew: Fix verbose mode

### LINKED ISSUE #4980: crew build breaking on patch
This is happening on trying to rebuild sommelier:

```
sommelier: Sommelier works by redirecting X11 and Wayland programs to the built-in ChromeOS wayland server.
Downloading source...
sh: 1: -o: not found
/usr/local/bin/crew:529:in `read': No such file or directory @ rb_sysopen - patch (Errno::ENOENT)
        from /usr/local/bin/crew:529:in `block in download'
        from /usr/local/bin/crew:511:in `chdir'
        from /usr/local/bin/crew:511:in `download'
        from /usr/local/bin/crew:858:in `build_package'
        from /usr/local/bin/crew:840:in `resolve_dependencies_and_build'
        from /usr/local/bin/crew:951:in `block in build_command'
        from /usr/local/bin/crew:947:in `each'
        from /usr/local/bin/crew:947:in `build_command'
        from /usr/local/bin/crew:1082:in `<main>'                                                 s�f��if85a3.diff�Uio�6�_1Mх/��%�I�d�                                           
     � �#���hPT��/iˊ7G�)PAIq���q�
                                 �rH�4D}�w�KҾ�&�C�d8t}�
��:/!�3+`�K�}����ǲl�B�<��nh�+8"j���$Y�mZ���v    ��Hz`��)±$���Gr�[�ei�*��������~�
                                                                                ��P0��#(Y�cF����D(iM�x��q<�}�o��h��tv� ��+S�SI��r<6E�4�ߌe�A��>��e��Nol��ִ�ET�Q�$#�D7㡃���پ���c%n@��������x�DgB,ˑin5��zט36ϰd�L��Ic�Js�E"e<w̮鸾�m�����jߋ8�*P��Za���4��1��U)=�&G%%�1�N6�籦뺦��R�9��k4�;x���g�J*7��wk� ���Zچ��I��<׶,�u�w�u���89����uz��h`�p["�Ă���JHA��9�C^ �x|���X�h��t�H@�*eUA�r�
```
Any ideas @uberhacker @supechicken

### PR DESCRIPTION
**Pull request by supechicken**
Fix #4980 
Tested on `x86_64`

### CODE REVIEW

**Comment by satmandu** [comment]
Did you test rebuilding any packages with this?

```crew reinstall -s pygments``` should do the trick...

**Comment by supechicken666** [comment]
> Did you test rebuilding any packages with this?
> 
> `crew reinstall -s pygments` should do the trick...

Yes, everything works well :)

File: crew [hunk 1]
```diff
@@ -644,26 +644,26 @@ def prepare_package (destdir)
...
 
 def strip_find_files (find_cmd, strip_option = "")
   # check whether crew should strip
-  return if CREW_NOT_STRIP || ENV['CREW_NOT_STRIP'] || !File.exist?("#{CREW_PREFIX}/bin/llvm-strip")
+  return if CREW_NOT_STRIP || ENV['CREW_NOT_STRIP'] || !File.exist?("#{CREW_PREFIX}/bin/strip")
 
   # run find_cmd and strip only ar or ELF files
   system "#{find_cmd} | xargs -r chmod u+w"
-  system "#{find_cmd} | xargs -r sh -c 'for i in \"$0\" \"$@\"; do case \"$(head -c 4 $i)\" in ?ELF|\!?ar) echo \"$i\";; esac ; done' | xargs -r llvm-strip #{strip_option}"
+  system "#{find_cmd} | xargs -r sh -c 'for i in \"$0\" \"$@\"; do case \"$(head -c 4 $i)\" in ?ELF|\!?ar) echo \"$i\";; esac ; done' | xargs -r strip #{strip_option}"
```
Comment (uberhacker) [inline crew:657]: It looks like you reverted @satmandu's recent commit.  Please pull the latest changes from master into your working branch.

Comment (supechicken) [reply crew]: Done

File: crew [hunk 2]
```diff
@@ -644,13 +644,13 @@ def prepare_package (destdir)
...
     # remove temporary files
-    "FileUtils.rm_rf ['dlistcut', '../dlist', '../filelist']#{@fileutils_verbose}"
+    FileUtils.rm_rf "['dlistcut', '../dlist', '../filelist']#{@fileutils_verbose}"
   end
 end
 
 def strip_find_files (find_cmd, strip_option = "")
   # check whether crew should strip
-  return if CREW_NOT_STRIP || ENV['CREW_NOT_STRIP'] || !File.exist?("#{CREW_PREFIX}/bin/llvm-strip")
+  return if CREW_NOT_STRIP || ENV['CREW_NOT_STRIP'] || !File.exist?("#{CREW_PREFIX}/bin/strip")
```
Comment (uberhacker) [inline crew:653]: This overwrites @satmandu's previous commit.

Comment (supechicken) [reply crew]: Done

File: crew [hunk 3]
```diff
@@ -843,8 +846,8 @@ def resolve_dependencies_and_build
   ensure
     #cleanup
     unless @opt_keep
-      FileUtils.rm_rf Dir.glob("#{CREW_BREW_DIR}/*")
-      FileUtils.mkdir_p "#{CREW_BREW_DIR}/dest" #this is a little ugly, feel free to find a better way
+      FileUtils.rm_rf "#{Dir.glob('#{CREW_BREW_DIR}/*')}#{@fileutils_verbose}"
```
Comment (uberhacker) [inline crew:849]: This syntax does not look correct.  The `#{...}` wrapper is for variables, not functions.

File: crew [hunk 4]
```diff
@@ -843,8 +846,8 @@ def resolve_dependencies_and_build
...
+      FileUtils.mkdir_p '#{CREW_BREW_DIR}/dest'#{@fileutils_verbose}" #this is a little ugly, feel free to find a better way
```
Comment (uberhacker) [inline crew:850]: Variables should always be enclosed by double quotes.  The single quotes make it literal.

**Comment by uberhacker** [comment]
Please test.  It's obvious from your code you haven't been doing this.

File: crew [hunk 5]
```diff
@@ -81,10 +81,12 @@ end
 
 if @opt_verbose then
   @fileutils_verbose = ', verbose: true'
```
Comment (uberhacker) [inline crew:83]: I'm thinking this shouldn't include the `,`.

**Comment by satmandu** [comment]
Please also make sure sommelier.rb builds successfully!

File: crew [hunk 6]
```diff
@@ -510,11 +512,11 @@ def download
   end
   Dir.chdir CREW_BREW_DIR do
     if ENV['CREW_CACHE_OPT']
-      FileUtils.mkdir_p CREW_CACHE_DIR
+      FileUtils.mkdir_p "#{CREW_CACHE_DIR}#{@fileutils_verbose}"
```
Comment (uberhacker) [inline crew:515]: If you get rid of the `,` in `@fileutils_verbose`, it would simply become:
```
FileUtils.mkdir_p CREW_CACHE_DIR, @fileutils_verbose
```

**Comment by supechicken** [comment]
> Please also make sure sommelier.rb builds successfully!

I can't get it working even using `crew 1.5.5` (#4883) :( 

Build successfully with #4889

File: crew [hunk 7]
```diff
@@ -673,9 +676,9 @@ def install_package (pkgdir)
     strip_find_files "find . -type f ! -iname '*\.bz2' ! -iname '*\.gz' ! -iname '*\.lha' ! -iname '*\.lz' ! -iname '*\.rar' ! -iname '*\.tar' ! -iname '*\.tbz' ! -iname '*\.tgz' ! -iname '\*.txz' ! -iname '*\.xz' ! -iname '*\.zip' -perm /111 -print | sed -e '/lib.*\.a$/d' -e '/lib.*\.so/d'"
 
     if Dir.exists? "#{pkgdir}/home" then
-      system "tar -cf #{@verbose} - ./usr/* ./home/* | (cd /; tar xp --keep-directory-symlink -f -)"
+      system "tar -c#{@verbose}f - ./usr/* ./home/* | (cd /; tar xp --keep-directory-symlink -f -)"
     else
-      system "tar -cf #{@verbose} - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)"
+      system "tar -#{@verbose}cf - ./usr/* | (cd /; tar xp --keep-directory-symlink -f -)"
```
Comment (uberhacker) [inline crew:681]: Can we make lines 679 and 681 consistent?

File: crew [hunk 8]
```diff
@@ -843,8 +846,8 @@ def resolve_dependencies_and_build
   ensure
     #cleanup
     unless @opt_keep
-      FileUtils.rm_rf Dir.glob("#{CREW_BREW_DIR}/*")
-      FileUtils.mkdir_p "#{CREW_BREW_DIR}/dest" #this is a little ugly, feel free to find a better way
+      FileUtils.rm_rf Dir.glob('#{CREW_BREW_DIR}/*'), verbose: @fileutils_verbose
```
Comment (uberhacker) [inline crew:849]: `"#{CREW_BREW_DIR}/*"` should have double instead of single quotes to expand the constant.

File: crew (hunk 7 above)
Comment (supechicken) [reply crew]: Done

**Comment by uberhacker** [comment]
Bump crew version in `lib/const.rb`?

**Comment by supechicken** [comment]
> Bump crew version in `lib/const.rb`?

Done

**Comment by satmandu** [comment]
This is still giving me problems with building new packages.

**Comment by uberhacker** [comment]
Can you please post your error?

**Comment by satmandu** [comment]
Sorry the issue was similar to before, with tar not creating the archive properly.

I think the problem is that you can't send ```*``` into an archive if you aren't in the folder where you want that to happen. 

```system "tar -c#{@verbose}Jf #{CREW_DEST_DIR}/#{pwd}/#{pkg_name} *``` won't work because that last glob parameter isn't looking inside  ```#{CREW_DEST_DIR}/```.  Also the ```#{pkg_name}``` tar.xz file is being put in ```#{CREW_DEST_DIR}/#{pwd}``` which equates to a subfolder of ```#{CREW_DEST_DIR}/```, which isn't where you want to be at all.

Something like this would probably be better:
```system "tar -c#{@verbose}Jf #{pwd}/#{pkg_name} #{CREW_DEST_DIR}/*``` 
or just ```system "tar c#{@verbose}Jf #{pwd}/#{pkg_name} #{CREW_DEST_DIR}/*``` 

I leave the dash off of tar because that's more forgiving, no?


Mind you I haven't tested those. ( didn't have time to generate packages that might be broken earlier, so I just used the earlier working version.)

**Pull request merged**